src/main/java/com/kolya/gym/controller/TraineeController.java

```diff
@@ -0,0 +1,164 @@
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK, Trainee returned"),
+            @ApiResponse(code = 400, message = "BAD_REQUEST, the request body or params are invalid or incorrect"),
+    })
+    @GetMapping("/{username}")
+    public ResponseEntity<?> getByUsername(@PathVariable("username") String username){
+        UUID transactionId = UUID.randomUUID();
+        logger.info("Transaction ID: {}, GET /trainees/{} was called", transactionId, username);
+        try{
+            userService.validateUsername(username);
+            Trainee trainee = traineeService.getByUsername(transactionId, username);
+            logger.info("Transaction ID: {}, 200 OK, Trainee was returned: {}",transactionId , trainee);
+            return ResponseEntity.status(HttpStatus.OK).body(trainee);
+        }catch (IllegalArgumentException e){
+            logger.info("Transaction ID: {}, 400 BAD_REQUEST, {}", transactionId, e.getMessage());
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
+        }
+    }
+
+
+    @ApiOperation(value = "Update trainee", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK, Trainee updated"),
+            @ApiResponse(code = 400, message = "BAD_REQUEST, the request body or params are invalid or incorrect"),
+    })
+    @PutMapping("/{username}")
+    public ResponseEntity<?> updateTrainee(@RequestBody TraineeDataUpdate traineeDataUpdate, @PathVariable("username") String username){
+        UUID transactionId = UUID.randomUUID();
+        logger.info("Transaction ID: {}, PUT /trainees/{} was called with body {}",transactionId, username,traineeDataUpdate);
+        try{
+            userService.validateUsername(username);
+            traineeDataUpdate.setUsername(username);
+            traineeDataUpdate.validate();
+            Trainee trainee = traineeService.update(transactionId, traineeDataUpdate);
+            logger.info("Transaction ID: {}, 200 OK, Trainee was updated: {}", transactionId, trainee);
+            return ResponseEntity.status(HttpStatus.OK).body(trainee);
+        }catch (IllegalArgumentException e){
+            logger.info("Transaction ID: {}, 400 BAD_REQUEST, {}", transactionId, e.getMessage());
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
+        }
+    }
+
+
+    @ApiOperation(value = "Delete trainee", response = ResponseEntity.class)
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "OK, Trainee deleted"),
+            @ApiResponse(code = 400, message = "BAD_REQUEST, the request body or params are invalid or incorrect"),
+    })
+    @DeleteMapping("/{username}")
+    public ResponseEntity<?> deleteByUsername(@PathVariable("username") String username){
+        UUID transactionId = UUID.randomUUID();
+        logger.info("Transaction ID: {}, DELETE /trainees/{} was called", transactionId, username);
```

Comment:
What kind of information does `transactionID ` provide us to help an investigation, if something goes wrong?